### PR TITLE
Rationalise sizing of HoistInputs and FormField

### PR DESCRIPTION
**Pull request by TomTirapani**
This branch includes some changes that will hopefully make sizing HoistInputs and FormFields more intuitive:

- HoistInputs can be split into two categories: "sized" and "unsized".
- Sized HoistInputs are now all exclusively sized using LayoutSupport. We previously had some sized inputs that were sized using named `width` & `height` props, and others using css.
- All sized inputs use `defaultProps` for reasonable, out-of-the-box widths. `TextArea` and `JsonInput` also have a default height. This means that all sized inputs always have an explicit size for their pertinent dimensions, which should mitigate unexpected sizes.
- When a sized input is inside a FormField, FormField unsets the width and flexes HoistInputs within it. It will also unset the height if the FormField has a specified height (but *only* if specified, allowing TextArea and JsonInput to use their default heights).
- Unsized HoistInputs (Checkbox, SwitchInput, RadioInput) do not have LayoutSupport, and do not expand to fill the dimensions of the FormField. Instead, these take their natural size from their content.

See issues #938 & #327
See Toolbox PR: https://github.com/exhi/toolbox/pull/149

### CODE REVIEW

File: desktop/cmp/input/DateInput.js [hunk 1]
```diff
@@ -79,10 +80,11 @@ export class DateInput extends HoistInput {
...
          * If undefined, control will not show time.
          */
-        timePrecision: PT.oneOf(['second', 'minute']),
+        timePrecision: PT.oneOf(['second', 'minute'])
+    };
 
-        /** Width of the control in pixels. */
-        width: PT.number
+    static defaultProps = {
+        width: 150
```
Comment (lbwexler) [inline desktop/cmp/input/DateInput.js]: we were moving away from specifying defaults with this mechanism

Comment (lbwexler) [reply desktop/cmp/input/DateInput.js]: instead, use withDefault in the render method.

File: desktop/cmp/form/FormField.js [hunk 2]
```diff
@@ -196,6 +213,19 @@ export class FormField extends Component {
             disabled: fieldModel && fieldModel.disabled,
             id: idAttr
         };
+
+        // Item should fill the available size of the FormField, unless dimensions are specified.
+        // Note: We explicitly set width / height to null to override defaults.
+        if (!item.props.width && !item.props.flex) {
```
Comment (lbwexler) [inline desktop/cmp/form/FormField.js]: both the checks on 219 should be isUndefined() to allow zero and null settings to be respected

**Comment by amcclain** [comment]
Already discussed with Tom, but adding here so we have a note on it and can pick it up again next week.  Saw some sizing irregularities on textarea and selects when switching between the various modes:

screen shot 2019-02-01 at 9 47 22 am
screen shot 2019-02-01 at 9 47 27 am
screen shot 2019-02-01 at 9 49 58 am

File: desktop/appcontainer/FeedbackDialog.js [hunk 3]
```diff
@@ -38,7 +38,9 @@ export class FeedbackDialog extends Component {
             items: [
                 textArea({
                     placeholder: 'Please enter your comments...',
-                    style: {height: 250, marginBottom: 2},
+                    width: null,
```
Comment (amcclain) [inline desktop/appcontainer/FeedbackDialog.js]: I am rushing a bit and reading this on github / not running the code, but I see a  few cases here and below where we are setting the width to null on these components.  Why is that?  It seems like, as a general goal, that should be a pretty unusual thing to need to do.

**Pull request merged**